components/jsx/preferences-modal/index.js

```diff
@@ -29,17 +40,38 @@ const positionModal = ({ eventTrigger, preferencesModal }) => {
 };
 
 const preferenceModalShowAndHide = ({ event, preferencesModal }) => {
-	if (!event.target) {
-		return;
-	}
-
 	preferencesModal.classList.toggle('n-myft-ui__preferences-modal--show');
 
 	if (preferencesModal.classList.contains('n-myft-ui__preferences-modal--show')) {
-		positionModal({ eventTrigger: event.target, preferencesModal });
+		positionModal({ event, preferencesModal });
+	} else {
+		// Remove existing errors when hiding the modal
+		renderError('');
```

Comment:
I think we need to specifc the `message` and `preferencesModal` here
```suggestion
		renderError({
			message: '',
			preferencesModal,
		});
```